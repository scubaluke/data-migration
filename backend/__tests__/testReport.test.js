require('regenerator-runtime/runtime')

const { newRecords, oldRecords } = require('./testDataset.js');
const dataStructureMethod = require('../reportGenerators/dataStructure')
const loopingMethod = require('../reportGenerators/looping')
const timeFunction = require('../utils/timeFunction')

describe('report method', () => {
  test('dataStructure should generate correct report', () => {
    const result = dataStructureMethod({ newRecords, oldRecords });
    // find missed records
    expect(result.missedRecords[0].name).toEqual('Missed in migration');
    // find corrupetd records
    expect(result.corruptedRecords[0].name).toEqual('corrupted');
    expect(result.corruptedRecords[1].email).toEqual('corruptEmail@gmail.com');
    // find new records
    expect(result.newlyCreatedRecords[0].name).toEqual('Newly added data');
    expect(result).toMatchSnapshot();
  });

  test('loop method should generate correct report', () => {
    const result = loopingMethod({ newRecords, oldRecords });
    // find missed records
    expect(result.missedRecords[0].name).toEqual('Missed in migration');
    // find corrupetd records
    expect(result.corruptedRecords[0].name).toEqual('corrupted');
    expect(result.corruptedRecords[1].email).toEqual('corruptEmail@gmail.com');
    // find new records
    expect(result.newlyCreatedRecords[0].name).toEqual('Newly added data');
    expect(result).toMatchSnapshot();
  });
});


test('time function should take in a function return its results as well as the run time', () => {
  const callback = () => {
    return { data: 'data passed through' }
  }
  const { runTime, data } = timeFunction({ callback })

  expect(runTime).toEqual('0.00');
  expect(data).toEqual('data passed through');
});