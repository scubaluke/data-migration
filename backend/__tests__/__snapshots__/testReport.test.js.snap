// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`report method dataStructure should generate correct report 1`] = `
Object {
  "corruptedRecords": Array [
    Object {
      "email": "myemail@gmail.com",
      "id": 1,
      "name": "corrupted",
    },
    Object {
      "email": "corruptEmail@gmail.com",
      "id": 12,
      "name": "steve",
    },
  ],
  "missedRecords": Array [
    Object {
      "email": "4someemail@gmail.com",
      "id": 14,
      "name": "Missed in migration",
    },
  ],
  "newlyCreatedRecords": Array [
    Object {
      "email": "youremail@gmail.com",
      "id": 2,
      "name": "Newly added data",
    },
  ],
}
`;

exports[`report method loop method should generate correct report 1`] = `
Object {
  "corruptedRecords": Array [
    Object {
      "email": "myemail@gmail.com",
      "id": 1,
      "name": "corrupted",
    },
    Object {
      "email": "corruptEmail@gmail.com",
      "id": 12,
      "name": "steve",
    },
  ],
  "missedRecords": Array [
    Object {
      "email": "4someemail@gmail.com",
      "id": 14,
      "name": "Missed in migration",
    },
  ],
  "newlyCreatedRecords": Array [
    Object {
      "email": "youremail@gmail.com",
      "id": 2,
      "name": "Newly added data",
    },
  ],
}
`;
